src/scale_impls/encode.rs

```diff
@@ -579,4 +598,48 @@ mod test {
         );
         assert_can_encode_to_type(Value::from_bytes([1]), SomeBytes(vec![1]));
     }
+
+    #[test]
+    fn can_encode_skipping_value_newtype_wrappers() {
+        #[derive(Encode, scale_info::TypeInfo)]
+        struct Foo {
+            inner: u32,
+        }
+
+        // The correct number of layers to match Foo
+        assert_can_encode_to_type(
+            Value::unnamed_composite(vec![Value::u128(32)]),
+            Foo { inner: 32 },
+        );
+        // One layer too many.
+        assert_can_encode_to_type(
+            Value::unnamed_composite(vec![Value::unnamed_composite(vec![Value::u128(32)])]),
+            Foo { inner: 32 },
+        );
+        // Two layers too many.
+        assert_can_encode_to_type(
+            Value::unnamed_composite(vec![Value::unnamed_composite(vec![
+                Value::unnamed_composite(vec![Value::u128(32)]),
+            ])]),
+            Foo { inner: 32 },
+        );
+    }
+
+    // As of https://github.com/paritytech/scale-value/issues/47, this test will take
+    // too long and time out. Keep the test to ensure that things do finish as expected!
+    #[test]
+    fn encoding_shouldnt_take_forever() {
```

Comment:
you could mark the test with `#[should_panic(expected = "some err")]` to make it more clear but good enough for me

Reply:
It's the opposite; it shouldn't panic unless it's failing :)

Reply:
>  // As of https://github.com/paritytech/scale-value/issues/47, this test will take
> // too long and time out. Keep the test to ensure that things do finish as expected!

Then I'm not understanding this comment :point_up: :monocle_face:

Reply:
ah make sense :)